### PR TITLE
HL-69

### PR DESCRIPTION
**Pull request by bazmatic**
[Link to ticket](https://dltx.atlassian.net/browse/HL-69)

- Configurable process for deploying to UAT/Local with new or pre-existing coin contracts
- Used `hardhat-deploy` plugin
- Includes an export of a `goerli` deployment ready for testing (see `exports/goerli.json`)

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+GOERLI_URL="https://goerli.infura.io/v3/663bcd65903948a6b53cd96866fc1a4a"
 MAINNET_URL=""
-ETHERSCAN_KEY=""
-PRIVATE_KEY=""
-HARDHAT_NETWORK="localhost"
-LOCAL_ETHER_BALANCE="1000000"
 
-FEE=100
-ORACLE_ADDRESS="0x0000000000000000000000000000000000000000"
+ETHERSCAN_KEY="K9FSU9Q3CGJBBSH9ZYWH7T4JCQIJ44QZ2R"
```
Comment (OptimusOpus) [inline .env.development:5]: Should we move these keys to Bitwarden?

File: contracts/MarketOracle.sol [hunk 2]
```diff
@@ -5,33 +5,58 @@ import "./IOracle.sol";
...
 
-    function checkResult(bytes32 marketId, bytes32 propositionId) external view returns (bool) {
-        require(propositionId != 0x0000000000000000000000000000000000000000000000000000000000000000, "getBinaryResult: Invalid propositionId");
+    function checkResult(
+        bytes32 marketId,
+        bytes32 propositionId
+    ) external view returns (bool) {
+        require(
+            propositionId !=
+                0x0000000000000000000000000000000000000000000000000000000000000000,
```
Comment (OptimusOpus) [inline contracts/MarketOracle.sol:21]: I am happy to keep  what is here, just thought i would mention that I think you could use ethers.constants.HashZero for something like this in the future

**Review by OptimusOpus — DISMISSED** [review]
LGTM

File: contracts/MarketOracle.sol (hunk 2 above)
Comment (bitcoinbrisbane) [reply contracts/MarketOracle.sol:21]: Wouldn't that be in the tests?

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
As discussed, roll back to underlying of the vaults underlying assets

File: contracts/MarketOracle.sol (hunk 2 above)
Comment (bazmatic) [reply contracts/MarketOracle.sol:21]: This is old code, not sure why it has come up as a change

Comment (bazmatic) [reply contracts/MarketOracle.sol:21]: Can't use ethers in solidity though @OptimusOpus

Comment (OptimusOpus) [reply contracts/MarketOracle.sol:21]: Oh whoops, my bad

**Review by OptimusOpus — APPROVED** [review]

**Review by xiangzhe0812 — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Pull request merged**